src/main/java/com/google/devtools/build/lib/skyframe/BzlLoadFunction.java

```diff
@@ -944,12 +944,12 @@ private static RepositoryMapping getRepositoryMapping(
     }
 
     if (key instanceof BzlLoadValue.KeyForBzlmod) {
-      if (repoName.equals(RepositoryName.BAZEL_TOOLS)) {
-        // Special case: we're only here to get the @bazel_tools repo (for example, for
-        // http_archive). This repo shouldn't have visibility into anything else (during repo
-        // generation), so we just return an empty repo mapping.
-        // TODO(wyv): disallow fallback.
-        return RepositoryMapping.ALWAYS_FALLBACK;
+      if (repoName.equals(RepositoryName.BAZEL_TOOLS) && isFileSafeForUninjectedEvaluation(key)) {
```

Comment:
better to just replace this `if` condition with a new allowlist in this PR. it's cleaner this way. (btw no need to have the `repoName` _and_ the `key` check

Reply:
I extracted the `PackageIdentifier` into a constant. If you prefer an actual (singleton) set or a function, I could change that.